### PR TITLE
Go package for smps

### PR DESCRIPTION
**Pull request by kanishkarj**
(I think) this creates a go package for SMPS.

### CODE REVIEW

**Comment by kanishkarj** [comment]
@leecalcote

File: smps.proto [hunk 1]
```diff
@@ -122,7 +125,7 @@ message ServiceMeshConfig {
 
     string smps_version = 1;
     string id = 2;
-    map labels = 3;
+    map<string, string> labels = 3;
```
Comment (leecalcote) [inline smps.proto:128]: is this our tagging / labeling system?

File: smps.proto [hunk 2]
```diff
@@ -2,6 +2,9 @@ syntax = "proto3";
 package SMPS;
 
 import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+option go_package = "spec;spec";
```
Comment (leecalcote) [inline smps.proto:7]: Should this be something like `layer5.smps.spec`?

File: smps.proto (hunk 1 above)
Comment (kanishkarj) [reply smps.proto:128]: Yes

File: smps.proto (hunk 2 above)
Comment (kanishkarj) [reply smps.proto:7]: We could do that, shall I ?

Comment (leecalcote) [reply smps.proto:7]: I suppose so. I think a better suggestion is `layer5io.service-mesh-performance-specification`

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module github.com/layer5io/smps
```
Comment (leecalcote) [inline go.mod:1]: Does this line have to be `module github.com/layer5io/service-mesh-performance-specification` or can we abbreviate with `smps`? The abbreviation is nice.

Comment (kanishkarj) [reply go.mod:1]: I stuck with the abbreviation because it's shorter.. it's easier to use that ways..

File: smps.proto (hunk 2 above)
Comment (kanishkarj) [reply smps.proto:7]: Should we switch to SMPS here too?

File: go.mod (hunk 3 above)
Comment (leecalcote) [reply go.mod:1]: Sounds great. I wasn't sure whether it was material (whether the reference needed to be exact).

File: smps.proto (hunk 2 above)
Comment (leecalcote) [reply smps.proto:7]: I would like that. I hesitate to suggest it in that my understanding is that this becomes a filesystem reference in `$GOPATH/SRC` when compiling the proto file (using `protoc`), but I'm not sure of all the considerations here. If `smps` works just as well, that's slick.

Comment (kanishkarj) [reply smps.proto:7]: I didn't get you exactly here.. but when I compile this proto file I get the output in here itself, In the spec folder.

Comment (leecalcote) [reply smps.proto:7]: I didn’t want to let the night slip by without helping get this merged. 

The consideration that I’m doing a poor job of articulating might be prevalent only when third parties are importing this package. I’m not sure.

**Pull request merged**